client/core/core_test.go

```diff
@@ -2714,6 +2728,243 @@ func TestTradeTracking(t *testing.T) {
+				standingOrders.booked.ID():              order.OrderStatusRevoked,  // booked, not canceled = assume revoked (may actually be executed)
+				standingOrders.bookedPendingCancel.ID(): order.OrderStatusCanceled, // booked pending canceled = assume canceled (may actually be revoked or executed)
+				standingOrders.executed.ID():            order.OrderStatusExecuted, // should not change
+				immediateOrders.epoch.ID():              order.OrderStatusExecuted, // preimage sent, not canceled = executed
+				immediateOrders.executed.ID():           order.OrderStatusExecuted, // should not change
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		// Track client orders in dc.trades.
+		dc.tradeMtx.Lock()
+		var pendingCancel *trackedTrade
+		dc.trades = make(map[order.OrderID]*trackedTrade)
+		for _, tracker := range tt.clientOrders {
+			dc.trades[tracker.ID()] = tracker
+			if tracker.cancel != nil {
+				pendingCancel = tracker
+			}
+		}
+		dc.tradeMtx.Unlock()
+
+		if pendingCancel != nil {
+			// Canceling the order again should fail.
+			rig.queueCancel(nil)
+			err = rig.core.Cancel(tPW, pendingCancel.ID().Bytes())
+			if err == nil {
+				t.Fatalf("no error for duplicate order cancellation")
+			}
+		}
+
+		// Queue order_status response if required for reconciliation.
+		if len(tt.orderStatuses) > 0 {
+			rig.ws.queueResponse(msgjson.OrderStatusRoute, func(msg *msgjson.Message, f msgFunc) error {
+				resp, _ := msgjson.NewResponse(msg.ID, tt.orderStatuses, nil)
+				f(resp)
+				return nil
+			})
+		}
+
+		// Reconcile tracked orders with server orders.
+		dc.reconcileTrades(tt.serverOrders)
+
+		dc.tradeMtx.RLock()
+		if len(dc.trades) != len(tt.expectOrderStatuses) {
+			t.Fatalf("%s: post-reconcileTrades order count mismatch. expected %d, got %d",
+				tt.name, len(tt.expectOrderStatuses), len(dc.trades))
+		}
+		for oid, tracker := range dc.trades {
+			expectedStatus, expected := tt.expectOrderStatuses[oid]
+			if !expected {
+				t.Fatalf("%s: unexpected order %v tracked by client", tt.name, oid)
+			}
+			tracker.mtx.RLock()
+			if tracker.metaData.Status != expectedStatus {
+				t.Fatalf("%s: client reported wrong order status %v, expected %v",
+					tt.name, tracker.metaData.Status, expectedStatus)
+			}
+			tracker.mtx.RUnlock()
+		}
+		dc.tradeMtx.RUnlock()
+
+		// Check if a previously canceled order existed; if the order is still
+		// active (Epoch/Booked status) and the cancel order is deleted, having
+		// been there for over 15 minutes since the cancel order's epoch ended.
+		if pendingCancel != nil {
+			pendingCancel.mtx.RLock()
+			status, stillHasCancelOrder := pendingCancel.metaData.Status, pendingCancel.cancel != nil
+			pendingCancel.mtx.RUnlock()
+			if status == order.OrderStatusBooked {
+				if stillHasCancelOrder {
+					t.Fatalf("%s: expected stale cancel order to be deleted for now-booked order", tt.name)
+				}
+				// Cancel order deleted. Canceling the order again should succeed.
+				rig.queueCancel(nil)
```

Comment:
This should be unnecessary, given the previously queued cancel response. If that previously queued response was used, this would return a `no response queued` error which will expose the fact that the previous cancel attempt visited the cancel route.